src/game/objects/har.c

```diff
@@ -779,20 +779,20 @@ void har_take_damage(object *obj, const str *string, float damage, float stun) {
     // Handle health changes
     if(h->health <= 0) {
         h->health = 0;
-        h->endurance = 0.0f;
     }
 
     if(!h->throw_duration) {
         log_debug("applying %f stun damage to %f", stun, h->endurance);
-        h->endurance -= stun;
+        h->endurance += stun;
     }
+
     if(h->endurance < 1.0f) {
         if(h->state == STATE_STUNNED) {
             // refill endurance
-            h->endurance = h->endurance_max;
-        } else {
-            h->endurance = 0.0f;
+            h->endurance = 0;
         }
+    } else if(h->endurance >= h->endurance_max) {
+        h->endurance = ((((h->endurance - h->endurance_max) / 256) * -2.5) - 60) * 256;
```

Comment:
A comment explaining what this is doing and what the expected result will be, would be useful?